src/main/java/duke/domain/Deadline.java

```diff
@@ -0,0 +1,99 @@
+package duke.domain;
+
+import java.time.LocalDateTime;
+import java.util.Objects;
+
+/**
+ * The type Deadline.
+ */
+public class Deadline extends Task {
+
+    private final LocalDateTime deadline;
+
+    private Deadline(String text, LocalDateTime deadline) {
+        super(text);
+        this.deadline = deadline;
+    }
+
+    @Override
+    public String toString() {
+        return String.format(
+                "[D] %s (by: %s)",
+                super.toString(),
+                this.deadline.format(FORMATTER));
+    }
+
+    @Override
+    public String exportString() {
+        return String.format(
+                "%s%s%s",
+                "D",
+                super.exportString(),
+                this.deadline.format(FORMATTER));
+    }
+
+    /**
+     * Of deadline.
```

Comment:
Maybe you want to elaborate on what the description "Of deadline" here means? :)